gnd/src/main/java/com/google/android/gnd/ui/offlinearea/OfflineAreaListAdapter.java

```diff
@@ -75,9 +90,42 @@ public void onClick(View v) {
   }
 
   @Override
-  public void onBindViewHolder(ViewHolder viewHolder, int position) {
-    viewHolder.binding.offlineAreaName.setText(offlineAreas.get(position).getId());
-    viewHolder.position = position;
+  public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
+    String areaName =
+        viewHolder.binding.getRoot().getContext().getString(R.string.offline_areas_unknown_area);
+
+    OfflineArea area = offlineAreas.get(position);
+    String id = area.getId();
+    LatLng center = area.getBounds().getCenter();
+
+    try {
+      List<Address> addresses =
+          viewHolder.geocoder.getFromLocation(center.latitude, center.longitude, 1);
+
+      if (addresses.isEmpty()) {
+        throw new AddressNotFoundException("No address found for area.");
+      }
+
+      Address address = addresses.get(0);
+
+      areaName =
+          Optional.ofNullable(address.getFeatureName())
+              .or(() -> Optional.ofNullable(address.getLocality()))
+              .or(() -> Optional.ofNullable(address.getSubAdminArea()))
+              .or(() -> Optional.ofNullable(address.getAdminArea()))
+              .or(() -> Optional.ofNullable(address.getCountryName()))
+              .orElse(areaName);
+    } catch (Exception e) {
```

Comment:
Can we please specify exactly which exception we are looking for? Is this catching the errors other than `AddressNotFoundException` raised above?

Reply:
done! 81c0b3f